lib/unix_util.cpp

```diff
@@ -80,7 +80,13 @@ int setenv(const char *name, const char *value, int overwrite) {
             }
             if (i!=envstrings.end()) {
                 // we allocated this string.  Reallocate it.
+		char *b=buf;
                 buf=(char *)realloc(buf,strlen(name)+strlen(value)+2);
+		if (!buf) {
+		    free(b);
+		    errno=ENOMEM;
+		    return -1;
```

Comment:
While I agree with the overall idea/implementation, these two lines look redundant to me (lines 98-101).

Reply:
Unless one favors early returns over redundant code, which I don't 😏

Reply:
I fail to see anything redundant. An sprintf to NULL is an uncontrolled abort. And that should be avoided.